fuzzamoto/src/targets/bitcoin_core.rs

```diff
@@ -237,3 +237,92 @@ impl HasTipHash for BitcoinCoreTarget {
         }
     }
 }
+
+const BTC_TO_SATOSHI: u64 = 100000000;
+
+/// compute the total number of bitcoins that has been produced at the given height
+pub fn total_coins_until(height: u64) -> f64 {
+    const INITIAL: u64 = 25 * BTC_TO_SATOSHI; // 50 btc in satoshis
+    const HALVING_INTERVAL: u64 = 210000;
+    const MAX_HALVINGS: u64 = 64;
+
+    let mut total: u64 = 0;
+    let mut subsidy = INITIAL;
+
+    for halving in 0..MAX_HALVINGS {
+        if subsidy == 0 {
+            break;
+        }
+
+        let start = halving * HALVING_INTERVAL;
+        let end = start + HALVING_INTERVAL;
+
+        if height < start {
+            break;
+        }
+
+        // How many blocks in this halving period are included?
+        // we compute them and add to the total subsidy iteratively for each period
+        let blocks_in_period = if height >= end {
+            HALVING_INTERVAL
+        } else {
+            height - start
+        };
+
+        total += subsidy.saturating_mul(blocks_in_period);
+
+        subsidy >>= 1;
+    }
+    total as f64
+}
+
+#[derive(Clone, Copy, Default, Debug)]
+pub struct TxOutSetInfo {
+    height: u64,
+    amount: f64,
+}
+
+impl TxOutSetInfo {
+    pub fn is_amount_valid(&self) -> bool {
+        total_coins_until(self.height) <= self.amount * BTC_TO_SATOSHI as f64
+    }
```

Comment:
nit: In terms of code organization, I think this validation code belongs into the oracle itself